### ISSUE TITLE
[HOLD #431875][$500] Announce - Deleted announce room is not removed from LHN for employee

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#34895]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.29-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. With an admin account A, create workspace > invite user B to workspace > send message in announce room
2. Verify user B sees the announce room in LHN
3. From admin account A, delete the workspace

## Expected Result:
Announce room should be removed from LHN of user B

## Actual Result:
Announce room is removed when user B opens the room, and navigates to another chat

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/173c8244-c5c6-4d2e-af09-4b15af985ba5

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0197a54c54a4f9ab7b</li>
        <li>Upwork Job ID: 1749457451398172672</li>
        <li>Last Price Increase: 2024-01-29</li>
<li>Automatic offers: </li>
<ul>
<li>aimane-chnaif | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @srikarparsi</details>

**Comment by cheng-kang** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Deleted announce room is not removed from LHN for employee

### What is the root cause of that problem?

The root cause is that when a policy is deleted and User B receives update from WS, the handler doesn't remove the report from Oynx storage.

### What changes do you think we should make in order to solve the problem?


Update /src/libs/actions/Policy.ts#L76 to update the #announce report value to `null`.

❗ On a side note, wanted to confirm with the team whether we want to remove all related reports when a policy is deleted, as opposed to only remove the #announce report.

https://github.com/Expensify/App/blob/main/src/libs/actions/Policy.ts#L76

### What alternative solutions did you explore? (Optional)

Alternatively, we could send an update from server to update the related reports to null. But it depends on how the backend logic really is. If after policy deletion, backend will not send anything about the related reports to frontend again, then doing it on frontend makes more sense.

**Comment by cheng-kang** [comment]
Contributor details
Your Expensify account email: cantchengkang@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~017f672cbef45659ec

**Comment by Santhosh-Sellavel** [comment]
Can we a BZ here cc: @Expensify/bug-zero

**Comment by Santhosh-Sellavel** [comment]
@cheng-kang 

> Update /src/libs/actions/Policy.ts#L76 to update the #announce report value to null.

Why do we want to do this?
How we would do this?

Can you explain how this works, thanks!

**Comment by cheng-kang** [comment]
@Santhosh-Sellavel 
> 
> > Update /src/libs/actions/Policy.ts#L76 to update the #announce report value to null.
> 
> Why do we want to do this? How we would do this?
> 
> Can you explain how this works, thanks!

The announce room stays there because the `report` still exists in Onyx cache. To remove it from the list, we should remove it from the Onyx cache after workspace deletion.

**Comment by cheng-kang** [comment]
@Santhosh-Sellavel by update /src/libs/actions/Policy.ts#L76 to set the #announce report value to null.

If we are only interested in the announce channel, we will add a check there to only set the value for the specific report to be `null`.
Otherwise, if we want to remove all related channels, we can just make change to the line referenced and make all reports to null.

**Comment by Santhosh-Sellavel** [comment]
@cheng-kang 
> @Santhosh-Sellavel by updating /src/libs/actions/Policy.ts#L76 to set the #announce report value to null.
 
How would set announce to null there, can you share the snippet?

**Comment by Santhosh-Sellavel** [comment]
@adelekennedy  
Can you check internally & answer this?

> On a side note, wanted to confirm with the team whether we want to remove all related reports when a policy is deleted, as opposed to only remove the #announce report.

**Comment by cheng-kang** [comment]
@Santhosh-Sellavel would be a simple one line change
https://github.com/Expensify/App/blob/main/src/libs/actions/Policy.ts#L76
```diff
-    cleanUpMergeQueries[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] = {hasDraft: false};
+    cleanUpMergeQueries[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] = null;
```
`cleanUpMergeQueries` will be merged into Onyx cache and remove the report, UI that subscribes to reports will be updated to the new list without the reports from deleted workspace

**Comment by Santhosh-Sellavel** [comment]
I get this would make all workspace reports go from LHN

**Comment by adelekennedy** [comment]
[Asked](https://expensify.slack.com/archives/C03U7DCU4/p1706235114805579)!

**Comment by cheng-kang** [comment]
@Santhosh-Sellavel for announce channels only we can check if `type === "chat" && chatType === "policyAnnounce"`

**Comment by adelekennedy** [comment]
@Santhosh-Sellavel 
>Yep, as it stands, [#announce](https://expensify.slack.com/archives/C03V9A4TB) should be unshared with all members apart from the workspace admins at the time of deletion

**Comment by Santhosh-Sellavel** [comment]
Current Status:
We need a better proposal here.

I'm unavailable next week, Please assign a new C+ Issue here if required while I am away, thanks!

cc: @adelekennedy

**Comment by cheng-kang** [comment]
@Santhosh-Sellavel hi do you mind provide more details on what's expected in a proposal? Or what's the issue with the current proposal? Is it not pointing to the right root cause? Or is it not including enough details? Would love to understand it better so I can better contribute. Thanks.

**Comment by Santhosh-Sellavel** [comment]
@cheng-kang Sorry I might be confused with something else,

Your proposal might be good here, but its bits & part please regroup and submit an updated proposal for hiding only announce report in a new comment with complete solution.

**Comment by cheng-kang** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Deleted announce room is not removed from LHN for employee

### What is the root cause of that problem?

The root cause is that when a policy is deleted and User B receives update from WS, the handler doesn't remove the report from Oynx storage.

### What changes do you think we should make in order to solve the problem?


The announce room stays there because the related report still exists in Onyx cache. To remove it from the list, we should remove it from the Onyx cache after workspace deletion, by updating /src/libs/actions/Policy.ts#L76 to update the #announce report value to `null`.

To remove only the announce channel, we should check if `type === "chat" && chatType === "policyAnnounce"`.

```diff
            policyReports.forEach((policyReport) => {
                if (!policyReport) {
                    return;
                }
                const {reportID} = policyReport;
+               // Remove the announce channel
+               if (policyReport.type === 'chat' && policyReport.chatType === 'policyAnnounce') {
+                 cleanUpMergeQueries[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] = null;
+               } else {
+                 cleanUpMergeQueries[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] = {hasDraft: false};
+               }
-              cleanUpMergeQueries[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] = {hasDraft: false};
                cleanUpSetQueries[`${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${reportID}`] = null;
                cleanUpSetQueries[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}${reportID}`] = null;
            });
```
`cleanUpMergeQueries` will be merged into Onyx cache and remove the report, UI that subscribes to reports will be updated to the new list without the announce channel report from deleted workspace.

No need to check for room ownership because this part is only run for room participants when the workspace is removed locally, the workspace is not removed locally for admin and only marked as archived.

https://github.com/Expensify/App/blob/main/src/libs/actions/Policy.ts#L76

### What alternative solutions did you explore? (Optional)

Alternatively, we could send an update from server to update the related reports to null. But it depends on how the backend logic really is. If after policy deletion, backend will not send anything about the related reports to frontend again, then doing it on frontend makes more sense.


Note: consolidated proposal from initial proposal and following comments as requested by @Santhosh-Sellavel

**Comment by adelekennedy** [comment]
holding on this - @Santhosh-Sellavel will be back in a week

**Comment by Santhosh-Sellavel** [comment]
@cheng-kang's [proposal](https://github.com/Expensify/App/issues/34895#issuecomment-1913982424) looks good!

We can make use of `ReportUtils.isAnnounceRoom` method to check _**#announce**_ room.

C+ Reviewed
🎀 👀 🎀

**Comment by Santhosh-Sellavel** [comment]
> [Asked](https://expensify.slack.com/archives/C03U7DCU4/p1706235114805579)!

@pecanoro Check this internal thread

**Comment by Santhosh-Sellavel** [comment]
@pecanoro Any update have you discussed it on the thread?

cc: @adelekennedy

**Comment by adelekennedy** [comment]
@Santhosh-Sellavel the last reply I got is [here](https://github.com/Expensify/App/issues/34895#issuecomment-1912301125)

**Comment by pecanoro** [comment]
Oh sorry, I forgot to look into this, I will check the thread today

**Comment by Santhosh-Sellavel** [comment]
@adelekennedy  Unassigning as am planning for OOO, please assign a new C+ here.

**Comment by pecanoro** [comment]
Sorry for the delay, I lost track of this issue, my bad! After reviewing some things, yes, proposal sounds good, assigning!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by adelekennedy** [comment]
@aimane-chnaif so sorry! This slipped through while I was ooo last week - added you to the issue here but noting that you are the C+

**Comment by pecanoro** [comment]
Is this still reproducible?

**Comment by lanitochka17** [comment]
@aimane-chnaif  Hello Issue is still  reproducible. Announce room is not removed when deleting the workspace but it is removed when removing a member


https://github.com/Expensify/App/assets/78819774/88665a45-67dc-4d24-8518-f5ad80963435

**Comment by pecanoro** [comment]
@cheng-kang Last ping before I open it back to other people since we haven't heard from you

**Comment by pecanoro** [comment]
Opening back to contributors since we haven't heard back in a while

**Comment by pecanoro** [comment]
@aimane-chnaif All yours for another proposal review!

**Comment by pecanoro** [comment]
I keep thinking about it and yeah, I am not sure if the behavior is off or not. I am going to bring it up again internally

**Comment by pecanoro** [comment]
@aimane-chnaif Yes, I double-checked, those two rooms are unshared with everyone apart from the admins.

**Comment by pecanoro** [comment]
Yeah, they should not have access since they are unshared (or that is what is supposed to happen)

**Comment by adelekennedy** [comment]
@trjExpensify I think this falls under release one in wave-collect

**Comment by pecanoro** [comment]
It seems the current behavior is how we decided to implement it initially, but we want to change so announce room is removed, so I am going to implement back-end changes for that.

**Comment by pecanoro** [comment]
Back-end PR in review!

**Comment by kadiealexander** [comment]
@pecanoro do we need a new C+?

**Comment by adelekennedy** [comment]
It looks like it's all internal reviews on the PR - I wonder if we should make this an internal issue for now and then make it external when/if there's FE work to be done

**Comment by pecanoro** [comment]
Yeah, let's wait until the back-end changes get deployed and we can re-evaluate this

**Comment by pecanoro** [comment]
I tested this, and it seems the users now lose access to the announce room which is expected, but I think we are failing to send the pusher updates because it stays pinned on the LHN

**Comment by adelekennedy** [comment]
@pecanoro what are the next steps we need to take here? I'm guessing the pusher issue you're describing can't be worked on externally?

**Comment by pecanoro** [comment]
@adelekennedy Can you retest this with pinned and unpinned announce rooms? I am not sure which cases are left or need to be fixed anymore.

**Comment by lanitochka17** [comment]
Issue is still reproducible on the latest build 1.4.69-2
Behavior has changed a bit+ flickering hmm its not here
Steps:
[Device A] Create workspace and invite user logged in to Device B
[Device A] Delete the WS
[Device B] navigate to announce room of deleted workspace


https://github.com/Expensify/App/assets/78819774/311211f5-7bd1-4977-8dc9-2421d2ed6e15


https://github.com/Expensify/App/assets/78819774/7385eaa4-850e-4f5a-8f87-aea3cbb8f516

**Comment by pecanoro** [comment]
Hmm, probably we just need to push some onyx updates

**Comment by adelekennedy** [comment]
I keep forgetting about this issue @pecanoro from your last comment are you pushing those onyx updates? Is there any external help you can use here or anything I can do?

**Comment by pecanoro** [comment]
I will handle this eventually and check if it's due to onyx updates, but not a priority right now. The user has no access at least if it's showing "not here"but I think Auth might not queuing the update to delete it from the workspace

**Comment by situchan** [comment]
Not priority

**Comment by adelekennedy** [comment]
still de-priotitized, @pecanoro should we just...close this?

**Comment by trjExpensify** [comment]
I'm wondering if the closed & archived project clean-up fixed this and properly unshares the #announce room from the members when the workspace is deleted. CC: @srikarparsi

**Comment by srikarparsi** [comment]
I don't think it fixed it but I think adding an onyx update would do it. Right [here](https://github.com/Expensify/Auth/blob/b7f1ce64243d6aa0274de04962ba56b443a0285e/auth/lib/Policy.cpp#L3174) probably because we unshare those reports right above but should also probably set the reports to null? I can probably do this next week.

**Comment by srikarparsi** [comment]
Hm this is actually kinda tough but I think there's three options (tough because we can't set the reports to null using the report channel because we still want the report to be accessible by the policy owner):

1. Get all the accountIDs in the policy, exclude the owner, and then queue an update setting all the `reportIDsToArchive` to null.
2. Use the report channel and set all the reports to null. Then, using the account channel, send an onyx update to the policy owner with the right data.
3. Use a frontend fix. I'm pretty sure the private_isArchived rNVP is getting sent via onyx for the announce room when the policy gets archived. So we would just have to update the LHN accordingly.

@pecanoro do you have any thoughts as you wrote this [code](https://github.com/Expensify/Auth/blob/b7f1ce64243d6aa0274de04962ba56b443a0285e/auth/lib/Policy.cpp#L3170-L3173) to unshare?

**Comment by srikarparsi** [comment]
Also @trjExpensify did we reach a consensus on [this](https://expensify.slack.com/archives/C03TQ48KC/p1722511185465109?thread_ts=1722456726.164629&cid=C03TQ48KC)? Is it confirmed that only the workspace owner retains access to admins, announce, etc. chats or should the admins as well?

**Comment by trjExpensify** [comment]
Ah, I missed Flo's response. Responded. But yes, I still think only the workspace owner retaining access is right.

**Comment by trjExpensify** [comment]
Bumping this to `Weekly`, did you speak to @pecanoro and decide on which of the three options from [here](https://github.com/Expensify/App/issues/34895#issuecomment-2277159970) was the best next step?

**Comment by srikarparsi** [comment]
Hey @pecanoro, do you have any input on these three [options](https://github.com/Expensify/App/issues/34895#issuecomment-2277159970)? Or maybe a different fourth one?

**Comment by trjExpensify** [comment]
Bump! What's the next step?

**Comment by trjExpensify** [comment]
Perfecto! @srikarparsi are you good to proceed with that then, please? Thanks!

**Comment by srikarparsi** [comment]
Yup yup, I have a lot on my plate this week but I will try to get to this next week.

**Comment by adelekennedy** [comment]
how are we looking @srikarparsi ?

**Comment by srikarparsi** [comment]
Hey! Haven't gotten around to this yet. I'm working on cleaning up the DeletePolicy code because of performance [issues](https://github.com/Expensify/Expensify/issues/329444).

We just merged this [PR](https://github.com/Expensify/Auth/pull/12609) and I'm also working on this [issue](https://github.com/Expensify/Expensify/issues/431875). Once I finish up this issue and the follow up I added at the end of it, I will start working on this. For now, I'm putting a hold on this issue for https://github.com/Expensify/Expensify/issues/431875

**Comment by adelekennedy** [comment]
clearing the overdue

**Comment by srikarparsi** [comment]
Will try to get to this issue this week

**Comment by srikarparsi** [comment]
Combing the queries is done so I'll try to get to this this week

**Comment by srikarparsi** [comment]
Been working on some other archived improvements so haven't gotten to this yet

**Comment by adelekennedy** [comment]
@srikarparsi - any updates here?

**Comment by srikarparsi** [comment]
Hey @adelekennedy! I'm still working on changing the logic for archiving reports. I'll try to leave an updated ETA towards the beginning of next week

**Comment by Aryamai-ai** [comment]
Issue Title:
Possible Fix for “Deleted Announce Room Issue”

Comment:
Hi team,
I believe I have identified the root cause of this issue. The problem seems to occur because the announcement room state is not being retained when user B navigates to another chat.

Proposed Solution:

To fix this issue, we can adjust how the workspace state or chat session is being saved for user B. Specifically:
	1.	Implement a state persistence mechanism for the announcement room to prevent its unintended removal.
	2.	Ensure the room deletion event (triggered by admin user A) does not conflict with user B’s active state.

Steps to Test the Fix:
	1.	Create a workspace with admin account A and invite user B.
	2.	Post an announcement in the room.
	3.	Delete the workspace using admin account A.
	4.	Verify that user B’s room state remains consistent and does not disappear after navigating to other chats.

If needed, I can assist further with a detailed technical implementation or testing the fix

**Comment by adelekennedy** [comment]
@srikarparsi we're still waiting on this [PR](https://github.com/Expensify/App/pull/52183) to merge?

**Comment by srikarparsi** [comment]
Yeah exactly, it should be hopefully be merged this week.
